pkg/datapath/linux/node.go

```diff
@@ -1366,6 +1366,52 @@ func (n *linuxNodeHandler) NodeNeighborRefresh(ctx context.Context, nodeToRefres
 	}
 }
 
+// NodeCleanNeighbors cleans all neighbor entries for the direct routing device
+// and the encrypt interface. It should be used when the agent changes the
+// state from `n.enableNeighDiscovery = true` to
+// `n.enableNeighDiscovery = false`.
+func (n *linuxNodeHandler) NodeCleanNeighbors() {
+	n.mutex.Lock()
```

Comment:
still pending - https://github.com/cilium/cilium/pull/15993#discussion_r625474856

Reply:
At the very least it should lock out concurrent attempts to list & delete neighbor entries which could otherwise result in a bunch of errors (assuming the caller can call into it from multiple goroutines).